mode_client/clients.py

```diff
@@ -224,6 +225,48 @@ def delete(self, space: str) -> None:
         self.request("DELETE", f"/spaces/{space}")
 
 
+class ModeDefinitionClient(ModeBaseClient):
+    def get(self, definition_token: str) -> Definition:
+        response = self.request("GET", f"/definitions/{definition_token}")
+
+        return Definition.parse_obj(response)
+
+    def list(
+        self, filter_: Optional[str] = None, tokens: Optional[List[str]] = None
+    ) -> List[Definition]:
+        params = {"filter": filter_, "tokens": tokens}
+        response = self.request("GET", "/definitions", params=params)
+        definitions = response["_embedded"]["definitions"]
+
+        return parse_obj_as(List[Definition], definitions)
+
+    def create(self, data: Dict[str, Any]) -> Definition:
+        response = self.request("POST", "/definitions", json={"definition": data})
+
+        return Definition.parse_obj(response)
+
+    def update(self, definition_token: str, data: Dict[str, Any]) -> Definition:
+        data = {k: v for k, v in data.items() if v is not None}
+        response = self.request(
+            "POST", f"/definitions/{definition_token}", json={"definition": data}
+        )
+
+        return Definition.parse_obj(response)
+
+    def delete(self, definition_token: str) -> None:
+        self.request("DELETE", f"definitions/{definition_token}")
+
+    def sync(
```

Comment:
wondering could we also have tests for these functions?

Reply:
yes, I will add some tests

Reply:
hmmm, okay then, let's skip the tests part for now for this PR 
but could we file another ticket to fix these tests? and maybe also add some basic tests for it?

Reply:
@treeforest456 sure, ticket: https://vouchinc.atlassian.net/browse/DE-4803